src/openfl/_internal/renderer/canvas/CanvasTilemap.hx

```diff
@@ -84,38 +83,14 @@ class CanvasTilemap {
 			}
 			
 			bitmapData = tileset.bitmapData;
-			if (bitmapData == null || !bitmapData.__prepareImage()) continue;
-			
-			if (bitmapData != cacheBitmapData) {
-				
-				if (bitmapData.image.buffer.__srcImage == null) {
-					
-					ImageCanvasUtil.convertToCanvas (bitmapData.image);
-					
-				}
-				
-				source = bitmapData.image.src;
-				cacheBitmapData = bitmapData;
-				
-			}
+			if (bitmapData == null || !bitmapData.__canBeDrawnToCanvas()) continue;
 			
 			context.globalAlpha = tilemap.__worldAlpha * alpha;
```

Comment:
I think I would move this right before the `__drawToCanvas` call like it's done in the other places in this PR.